mantidproject/mslice
# Implement first cut at CLI

It would be advisable to merge #68  before testing this pull request.

Implement Command line interface for mslice. Because the plotting of the cuts was handled by the presenter it was tedious to expose to the CLI. The Actual plotting has been moved to a `CutPlottter` class.
The entrie conents of `matplotlib.pyplot` was copied to `plotting.pyplot`. A few functions were redefined and a few others were omitted. The omitted functions returned by the function `cli.get_unincluded_functions`. The redefined functions mimic the interface of `matplotlib.pyplot` and are returned by `cli.redefinded_functions`

All custom functions are defined in `cli._mslice_commands` The file has is divided into sections. The first part under the header `#Helper tools` defines convenience functions used throughout the file.
 The next part (is a single line) under header `#Mantid Tools` imports  mantid `Algorithms` exposed under cli for the users convenience (currently `mtd`, `Load`, `ConvertUnits`, `RenameWorkspace`)
The following 3 sections are Projecting, Slicing and Cutting respectively.

**To test:** 
1. Start MSlice with a console (this can be achieved by merging in the branch `startup_scripts` and running `start_mslice.sh`/`start_mslice.bat`
2.Verify that taking cuts via the GUI still works as expected.
3. from the command line run`import cli`
4. Try loading a workspace via `cli.Load`
5.  Verify that that the docstring of `cli.get_projection` provides enough guidance for a user to be able to use it. You can view the docstring by executing `help(cli.get_projection)`

6. Verify the readability of the docstrings of  `cli.get_slice` and `cli.plot_slice` 
7. Using projection made in 5 test the functions   `cli.get_slice` and `cli.plot_slice` 

8. Verify readablility of docstrings of `cli.get_cut_xye` and `cli.plot_cut_xye`
9. 7. Using projection made in 5 test the functions `cli.get_cut_xye` and `cli.plot_cut_xye`

10. test that the pyplot api functions are accessible via `cli`. if #68 is not merged then calling `cli.figure()` may result in erratic behaviour for subsequent use of the MSLice GUI.
Fixes #55
